### PR TITLE
cpp object detection demo: faceboxes fixes

### PR DESCRIPTION
**Pull request by ivanvikhrev**
Store anchors and do calculations in float (at first did it in int, it was a rash decision), plus small code style fixes.

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/common/cpp/models/src/detection_model_faceboxes.cpp [hunk 1]
```diff
@@ -98,14 +104,14 @@ void calculateAnchors(std::vector<ModelFaceBoxes::Anchor>* anchors, const std::v
 
     for (auto cy : dense_cy) {
         for (auto cx : dense_cx) {
-            anchors->push_back({ static_cast<int>(cx - 0.5f * skx), static_cast<int>(cy - 0.5f * sky),
-                static_cast<int>( cx + 0.5f * skx), static_cast<int>(cy + 0.5f * sky) });  // left top right bottom
+            anchors.push_back({ static_cast<float>(cx - 0.5f * skx), static_cast<float>(cy - 0.5f * sky),
+                static_cast<float>( cx + 0.5f * skx), static_cast<float>(cy + 0.5f * sky) });  // left top right bottom
```
Comment (IRDonch) [inline demos/common/cpp/models/src/detection_model_faceboxes.cpp:108]: Seems like the casts shouldn't be necessary anymore.

**Comment by ivanvikhrev** [comment]
Jenkins please retry a build

**Review by Wovchena — APPROVED** [review]
Jenkins ok to test

**Comment by Wovchena** [comment]
Jenkins ok to test

**Pull request merged**